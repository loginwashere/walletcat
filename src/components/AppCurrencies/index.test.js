import React from 'react'
import renderer from 'react-test-renderer'
import {
  currencySeeder,
  userCurrencySeeder
} from '../../../server/seeds'
import { AppCurrencies } from '.'

describe('components:AppCurrencies:', () => {
  it('renders without crashing and without data', () => {
    const currency = currencySeeder.items[0]
    const currencies = { [currency.id]: currency }
    const currencyIds = [currency.id]
    const userCurrencies = {}
    const userCurrenciesIdsByCurrencyId = {}
    const dispatch = jest.fn()
    const tree = renderer.create(
      <AppCurrencies currencies={currencies}
                     currencyIds={currencyIds}
                     userCurrencies={userCurrencies}
                     userCurrenciesIdsByCurrencyId={userCurrenciesIdsByCurrencyId}
                     dispatch={dispatch} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without crashing and with data', () => {
    const currency = currencySeeder.items[0]
    const currencies = { [currency.id]: currency }
    const currencyIds = [currency.id]
    const userCurrency = userCurrencySeeder.items[0]
    const userCurrencies = { [userCurrency.id]: userCurrency }
    const userCurrenciesIdsByCurrencyId = { [currency.id]: userCurrency.id }
    const page = 1
    const pages = { [page]: { ids: [currency.id] } }
    const dispatch = jest.fn()

    const tree = renderer.create(
      <AppCurrencies currencies={currencies}
                     currencyIds={currencyIds}
                     userCurrencies={userCurrencies}
                     userCurrenciesIdsByCurrencyId={userCurrenciesIdsByCurrencyId}
                     dispatch={dispatch}
                     page={page}
                     pages={pages} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

