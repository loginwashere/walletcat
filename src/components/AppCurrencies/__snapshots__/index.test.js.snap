exports[`components:AppCurrencies: renders without crashing and with data 1`] = `
<div>
  <h1>
    App Currencies
  </h1>
  <div
    className="list-group">
    <li
      className="list-group-item">
      <button
        className="pull-right btn btn-xs btn-default"
        disabled={false}
        onClick={[Function]}
        type="button">
        Remove from My Currencies
      </button>
      <h4
        className="list-group-item-heading">
        AED
      </h4>
      <p
        className="list-group-item-text">
        United Arab Emirates Dirham
      </p>
    </li>
  </div>
  <ul
    className="pager">
    <li
      className="disabled previous"
      style={undefined}>
      <a
        action="push"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
        tabIndex={-1}>
        Previous Page
      </a>
    </li>
    <li
      className="disabled next"
      style={undefined}>
      <a
        action="push"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
        tabIndex={-1}>
        Next Page
      </a>
    </li>
  </ul>
</div>
`;

exports[`components:AppCurrencies: renders without crashing and without data 1`] = `
<div>
  <h1>
    App Currencies
  </h1>
  <ul
    className="pager">
    <li
      className="disabled previous"
      style={undefined}>
      <a
        action="push"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
        tabIndex={-1}>
        Previous Page
      </a>
    </li>
    <li
      className="disabled next"
      style={undefined}>
      <a
        action="push"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
        tabIndex={-1}>
        Next Page
      </a>
    </li>
  </ul>
</div>
`;
